server/ocsp.go

```diff
@@ -446,13 +446,33 @@ func (srv *Server) NewOCSPMonitor(config *tlsConfigKind) (*tls.Config, *OCSPMoni
 				if len(s.VerifiedChains) == 0 {
 					return fmt.Errorf("%s client missing TLS verified chains", kind)
 				}
+
 				chain := s.VerifiedChains[0]
-				resp, err := ocsp.ParseResponseForCert(oresp, chain[0], issuer)
+				leaf := chain[0]
+				parent := issuers[len(issuers)-1]
+
+				resp, err := ocsp.ParseResponseForCert(oresp, leaf, parent)
 				if err != nil {
 					return fmt.Errorf("failed to parse OCSP response from %s client: %w", kind, err)
 				}
-				if err := resp.CheckSignatureFrom(issuer); err != nil {
-					return err
+				if resp.Certificate == nil {
```

Comment:
Reference on why need to check here that it is `nil`: https://github.com/nats-io/nats-server/blob/91d6531b8bdbcee0fdc1d2a4810712cdf6f76a7b/vendor/golang.org/x/crypto/ocsp/ocsp.go#L401-L406
```
// CheckSignatureFrom checks that the signature in resp is a valid signature
// from issuer. This should only be used if resp.Certificate is nil. Otherwise,
// the OCSP response contained an intermediate certificate that created the
// signature. That signature is checked by ParseResponse and only
// resp.Certificate remains to be validated.
```